### PR TITLE
tests: add unit test cases for reporting metrics

### PR DESCRIPTION
**Pull request by Mohammad-Tayyab-Frequenz**
This PR adds comprehensive unit test coverage for the reporting metrics module, which provides energy flow calculations for hybrid power systems including production, consumption, battery storage, and grid interactions.

- Introduces 9 test cases covering all key functions in the reporting_metrics module
- Tests edge cases including sign conventions, missing values, clipping behavior, and error conditions
- Validates warning emissions and error handling for invalid inputs

### CODE REVIEW

File: tests/test_reporting_metrics.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+
+from frequenz.lib.notebooks.reporting.metrics import reporting_metrics as metrics
+
+
+def test_asset_production_respects_sign_and_missing_values() -> None:
+    """Positive flag controls sign inversion while NaNs become zeros."""
```
Comment (cwasicki) [inline tests/test_reporting_metrics.py]: How do we know that we can turn NaNs into zeroes?

File: tests/test_reporting_metrics.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    production = pd.Series([10, -5, None], index=pd.RangeIndex(3))
```
Comment (cwasicki) [inline tests/test_reporting_metrics.py:17]: Are `None`s turned into `NaN`s?

File: tests/test_reporting_metrics.py (hunk 1 above)
Comment (Mohammad-Tayyab-Frequenz) [reply tests/test_reporting_metrics.py]: Yes, I remember now we discussed that we should not convert to zero. 
I will fix it in the code.

File: tests/test_reporting_metrics.py (hunk 2 above)
Comment (Mohammad-Tayyab-Frequenz) [reply tests/test_reporting_metrics.py:17]: Yes, but with the previous comment the test case would change.

File: tests/test_reporting_metrics.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+def test_asset_production_respects_sign_and_missing_values() -> None:
+    """Positive flag controls sign inversion and NaNs stay missing."""
+    production = pd.Series([10, -5, None], index=pd.RangeIndex(3))
+
+    positive = metrics.asset_production(production, production_is_positive=True)
+    expected_positive = pd.Series([10.0, 0.0, float("nan")], index=production.index)
+    assert_series_equal(positive, expected_positive)
+
+    negative = metrics.asset_production(production, production_is_positive=False)
```
Comment (cwasicki) [inline tests/test_reporting_metrics.py:23]: Not for this PR, but I feel we should remove the `production_is_positive` flag and stick to PSC wherever possible. And in the other cases we can just explicitly reverse the sign, correct?

File: tests/test_reporting_metrics.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    assert_series_equal(result, expected)
+
+
+def test_grid_feed_in_excludes_energy_stored_in_battery() -> None:
+    """Grid feed-in equals production surplus minus what is stored in the battery."""
+    production = pd.Series([-8, -3], index=pd.RangeIndex(2))
+    consumption = pd.Series([2, 1], index=production.index)
+    battery = pd.Series([5, 10], index=production.index)
+
+    result = metrics.grid_feed_in(production, consumption, battery)
```
Comment (cwasicki) [inline tests/test_reporting_metrics.py:55]: In future we might generalize this to storage instead of battery only.

**Review by cwasicki — APPROVED** [review]
LGTM, thank you!

File: tests/test_reporting_metrics.py (hunk 3 above)
Comment (Mohammad-Tayyab-Frequenz) [reply tests/test_reporting_metrics.py:23]: Yes, we can do that.

File: tests/test_reporting_metrics.py (hunk 4 above)
Comment (Mohammad-Tayyab-Frequenz) [reply tests/test_reporting_metrics.py:55]: Got it. I will make note of it as well.
